src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -42,6 +44,9 @@
 	public String texture;
 	public String customModelName;
 
+	public Map<String, Procedure> customProperties;
+	public Map<Map<String, Float>, ModelEntry> modelsMap;

```

Comment:
Map with Map as a key? Not sure how wise this is, usually this means issue with data model if this is needed..

Reply:
Available templates generate properly with this, not sure what do you mean by **data model**. I could use a string type keys (like `property1=false,property2=354`), but how do I turn it into an proper map in freemarker?

Reply:
I know they generate properly, but using a map as a key is an antipattern.

But maybe it is needed here, it is just that it is not common and should be avoided when possible

Reply:
:heavy_check_mark: Fixed, seems to work fine